aiidalab_ispg/spectrum_analysis.py

```diff
@@ -224,3 +241,389 @@ def _observe_disabled(self, change):
+
+        self.yield_slider = ipw.FloatSlider(
+            min=0.01,
+            max=1,
+            step=0.01,
+            value=1,
+            description="Quantum yield",
+            continuous_update=True,
+            disabled=False,
+        )
+        self.yield_slider.observe(self.handle_slider_change, names="value")
+
+        self.flux_data = self.read_in_actinic()
+
+        self.number = ipw.HTML(
+            description=r"$$Photolysis rate constant (s^{-1})$$ =",
+            style={"description_width": "initial"},
+            disabled=True,
+        )
+
+        bokeh_tools = "pan,wheel_zoom,box_zoom,reset,save"
+        figure_size = {
+            "sizing_mode": "stretch_width",
+            "height": 400,
+            "max_width": 400,
+        }
+        self.figure = self._init_figure(tools=bokeh_tools, **figure_size)
+        self.figure.layout = ipw.Layout(overflow="initial")
+
+        super().__init__(
+            children=[self.flux_toggle, self.yield_slider, self.number, self.figure]
+        )
+        layout = ipw.Layout(justify_content="flex-start")
+
+    # **********************************************************************
+    def _init_figure(self, *args, **kwargs) -> BokehFigureContext:
+        """Initialize Bokeh figure. Arguments are passed to bokeh.plt.figure()"""
+        figure = BokehFigureContext(plt.figure(*args, **kwargs))
+        f = figure.get_figure()
+        f.xaxis.axis_label = r"$$λ (nm)$$"
+        f.yaxis.axis_label = r"$$j (s^{-1} nm^{-1})$$"
+        f.x_range = Range1d(280, 400)
+        f.y_range = Range1d(0, 3e-5)
+
+        f.extra_y_ranges = {"V": Range1d(start=1.0, end=1e15)}
+        f.extra_y_scales = {"V": LogScale()}
+        f.add_layout(
+            LogAxis(
+                y_range_name="V",
+                axis_label=r"$$ F \hspace{1mm} (quanta \hspace{1mm} cm^{-2} \hspace{1mm} s^{-1} \hspace{1mm} nm^{-1})$$",
+            ),
+            "right",
+        )
+
+        return figure
+
+    # **********************************************************************
+
+    # **********************************************************************
+    # Observing changes in widget and spectrum data, update j plot for each change
+
+    @traitlets.observe("spectrum_data")
+    def _observe_spectrum_data(self, change):
+        """
+        Observe changes to the spectrum data and update the J plot accordingly.
+        Check that fluxdata overlaps with the spectrum data.
+
+        :param change: The change object containing the old and new values of the spectrum data.
+        :type change: dict
+        """
+        self.disabled = True
+        if change["new"] is None or len(change["new"]) == 0:
+            self.reset()
+            return
+
+        flux_min = min(self.flux_data[0])
+        spectrum_max = max(self.spectrum_data[0])
+
+        # Check end of spectrum data overlaps with flux data
+        if spectrum_max >= flux_min:
```

Comment:
In principle, you should also check that the other side of the interval to be on the safe side. :-) 

```suggestion
        if spectrum_max >= flux_min and spectrum_min < flux_max:
```